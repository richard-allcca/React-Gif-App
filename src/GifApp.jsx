import React, { useState } from 'react'
import { AddCategory, GifGrid } from './components/';


const GifApp = () => {

  const [ categories, setCategories ] = useState('')

  const onAddCategory = (newValue) => {

    // TODO: format newValue to be a valid category name
    // if (categories.includes(newValue)) return;

    setCategories(newValue)
  }


  return (
    <>
      <h1>Gif-app</h1>

      <AddCategory onAddCategory={(newValue) => onAddCategory(newValue)} />

      <GifGrid categoria={categories} />
      {/* {categories.map((item, id) => {
        return <GifGrid key={id} categoria={item} />
      })} */}

    </>
  )
}

export default GifApp;