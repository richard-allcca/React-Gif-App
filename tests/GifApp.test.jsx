import { render } from "@testing-library/react";
import GifApp from "../src/GifApp";


describe("Pruebas en <GifApp/>", () => {
   test("Debe mostrar un snapshop del componente", () => {
      const { container } = render(<GifApp />);

      expect(container).toMatchSnapshot();
   });

   test("Debe mostrar una lista de gifs", () => {
      // TODO: pendiente de implementar
   });
});
