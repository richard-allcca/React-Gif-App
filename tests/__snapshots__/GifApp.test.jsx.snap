// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifApp/> Debe mostrar un snapshop del componente 1`] = `
<div>
  <h1>
    Gif-app
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      Agregar
    </button>
  </form>
  <h3 />
  <p>
    Loading...
  </p>
  <div
    class="card-grid"
  />
</div>
`;
